filesystem/src/main/java/org/frankframework/filesystem/LocalFileSystem.java

```diff
@@ -273,8 +280,45 @@ public Date getModificationTime(Path f) throws FileSystemException {
 
 	@Override
 	@Nullable
-	public Map<String, Object> getAdditionalFileProperties(Path f) {
-		return null;
+	public Map<String, Object> getAdditionalFileProperties(Path file) throws FileSystemException {
+		try {
+			if (!Files.exists(file)) return null;
+			UserDefinedFileAttributeView userDefinedAttributes = Files.getFileAttributeView(file, UserDefinedFileAttributeView.class);
+			List<String> attributeNames = userDefinedAttributes.list();
+			if (attributeNames == null || attributeNames.isEmpty()) return null;
+			String attrSpec = "user:" + String.join(",", attributeNames);
+			Map<String, Object> result = new LinkedHashMap<>();
+			Files.readAttributes(file, attrSpec)
+					.forEach((name, value) -> result.put(name, readAttributeValue(value)));
+			return result;
+		} catch (UnsupportedOperationException e) {
+			return null;
+		} catch (IOException e) {
+			throw new FileSystemException(e);
+		}
+	}
+
+	private String readAttributeValue(Object attributeValue) {
+		if (attributeValue instanceof byte[] bytes) {
+			return new String(bytes);
+		} else if (attributeValue instanceof ByteBuffer buffer) {
+			return new String((buffer).array());
+		} else {
+			return attributeValue.toString();
+		}
+	}
+
+	@Override
+	public void setCustomFileAttribute(@Nonnull Path file, @Nonnull String key, @Nonnull String value) {
+		try {
+			if (!Files.exists(file)) {
+				Files.createFile(file);
+			}
```

Comment:
I didn't expect this method to create a file.. feels like a side-effect.

Reply:
It's the only way though to set extended attributes on the file, because unlike in S3 I cannot add them to the in-memory representation of the file.
If I set properties after creating the file object, then this will impact the primary use-case of setting these properties  for S3.

Setting them for LocalFileSystem was just to see if I could.

Reply:
Meest waarschijnlijk wordt deze method aangeroepen voordat de file echt bestaat op het fysieke filesystem.
Ik kan:
1. Of het side-effect documenteren
2. Of dit weer uit het LocalFileSystem halen.
(Geeft niet om dit niet te doen in LocalFileSystem, dat was geen requirement, maar een extra om te kijken of het kan).

Reply:
Zeker documenteren. Gaat dit goed als je een file verplaatst?
(Misschien dat de parameter values in de metadata map gezet kunnen worden, dan heb je mogelijk dit probleem niet?)

Reply:
Het OS managed deze metadata dus het OS managed ook dat de metadata mee verhuist met het bestand wanneer je het verplaatst of kopieert.